libcloud/test/storage/test_oss.py

```diff
@@ -0,0 +1,880 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import unittest
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+try:
+    from lxml import etree as ET
+except ImportError:
+    from xml.etree import ElementTree as ET
+
+from libcloud.utils.py3 import b
+from libcloud.utils.py3 import httplib
+from libcloud.utils.py3 import urlparse
+from libcloud.utils.py3 import parse_qs
+from libcloud.utils.py3 import PY3
+from libcloud.common.types import InvalidCredsError
+from libcloud.common.types import MalformedResponseError
+from libcloud.storage.base import Container, Object
+from libcloud.storage.types import ContainerDoesNotExistError
+from libcloud.storage.types import ContainerError
+from libcloud.storage.types import ContainerIsNotEmptyError
+from libcloud.storage.types import InvalidContainerNameError
+from libcloud.storage.types import ObjectDoesNotExistError
+from libcloud.storage.types import ObjectHashMismatchError
+from libcloud.storage.drivers.oss import OSSConnection
+from libcloud.storage.drivers.oss import OSSStorageDriver
+from libcloud.storage.drivers.oss import CHUNK_SIZE
+from libcloud.storage.drivers.dummy import DummyIterator
+from libcloud.test import StorageMockHttp, MockRawResponse  # pylint: disable-msg=E0611
+from libcloud.test import MockHttpTestCase  # pylint: disable-msg=E0611
+from libcloud.test.file_fixtures import StorageFileFixtures  # pylint: disable-msg=E0611
+from libcloud.test.secrets import STORAGE_OSS_PARAMS
+
+
+class OSSConnectionTestCase(unittest.TestCase):
+    def setUp(self):
+        self.conn = OSSConnection('44CF9590006BF252F707',
+                                  'OtxrzxIsfpFjA7SwPzILwy8Bw21TLhquhboDYROV')
+
+    def test_signature(self):
+        expected = b('26NBxoKdsyly4EDv6inkoDft/yA=')
+        headers = {
+            'Content-MD5': 'ODBGOERFMDMzQTczRUY3NUE3NzA5QzdFNUYzMDQxNEM=',
+            'Content-Type': 'text/html',
+            'Expires': 'Thu, 17 Nov 2005 18:49:58 GMT',
+            'X-OSS-Meta-Author': 'foo@bar.com',
+            'X-OSS-Magic': 'abracadabra',
+            'Host': 'oss-example.oss-cn-hangzhou.aliyuncs.com'
+        }
+        action = '/oss-example/nelson'
+        actual = OSSConnection._get_auth_signature('PUT', headers, {},
+                                                   headers['Expires'],
+                                                   self.conn.key,
+                                                   action,
+                                                   'x-oss-')
+        self.assertEqual(expected, actual)
+
+
+class ObjectTestCase(unittest.TestCase):
+    def test_object_with_chinese_name(self):
+        driver = OSSStorageDriver(*STORAGE_OSS_PARAMS)
+        obj = Object(name=u'中文', size=0, hash=None, extra=None,
```

Comment:
Changing this to use `libcloud.utils.py3.u` wrapper function should fix the test failure under Python 3.2.

So:

```python

from libcloud.utils.py3 import u
...

obj = Object(name=u('中文'), size=0, hash=None, extra=None,
...
```